herbert_core/classes/data_loaders/@rundata/rm_masked.m

```diff
@@ -1,30 +1,58 @@
-function [S_m,Err_m,det_m]=rm_masked(this)
-% method removes failed (NaN or Inf) data from the data array and deletes
+function [S_m,Err_m,det_m]=rm_masked(obj,ignore_nan,ignore_inf)
+% Method removes failed (NaN or Inf) data from the data array and deletes
 % detectors, which provided such signal
 %
-if isempty(this.S)||isempty(this.ERR)||isempty(this.det_par)
+% Input:
+% obj -- initialized rundata object to check for failied detectors
+% Optional:
+% ignore_nan -- mask detectors with Nan signal. If absent, assumed true
+% ignore_inf -- mask detectors with Inf signal. If absent, assumed true
+%
+% providing both ignore_nan and ignore_inf equal to false, disables masking
+%
+if isempty(obj.S)||isempty(obj.ERR)||isempty(obj.det_par)
     error('RUNDATA:rm_masked',' signal, error and detectors arrays have to be defined\n');
 end
-if any(size(this.S)~=size(this.ERR))||(size(this.S,2)~=numel(this.det_par.x2))
+if any(size(obj.S)~=size(obj.ERR))||(size(obj.S,2)~=numel(obj.det_par.x2))
     error('RUNDATA:rm_masked',' signal error and detectors arrays are not consistent\n');
 end
+if ~exist('ignore_nan','var')
+    ignore_nan = true;
+end
+if ~exist('ignore_inf','var')
+    ignore_inf = true;
+end
+
 
-index_masked = (isnan(this.S)|(isinf(this.S))); % masked pixels
-line_notmasked= ~any(index_masked,1);           % masked detectors (for any energy)
+if ignore_nan && ignore_inf
+    index_masked = isnan(obj.S)| isinf(obj.S); % masked pixels
+elseif ignore_nan
+    index_masked = isnan(obj.S);
+elseif ignore_inf
+    index_masked = isinf(obj.S);
+else
+    S_m= obj.S;
```

Comment:
It would be clearer to move the 'early return' logic into a separate if block before the `index_masked` logic, i.e.
```
if ~ignore_nan && ~ignore_inf
  ...set stuff...
  return
end

if ignore_nan && ignore_inf
   index_masked = isnan(obj.S)| isinf(obj.S); % masked pixels
   ...
```

It's easy to miss a `return` buried in a multi-branch if.

Reply:
probably. Difficult to judge. Will try to change -- it leaves one of the if branches hanging, which is not obvious too.